File: python-prov/src/default_namespace.py
import datetime
import sys

import prov.dot
from prov.model import ProvDocument
import prov.graph

from test_case import TestCase


class DefaultNamespace(TestCase):

    def __init__(self):
        self.filename = "default_namespace"

    def create_document(self):
        document = ProvDocument()
        document.set_default_namespace("https://default.org/")
        document.add_namespace("ex", "https://example.org/")
        document.entity("ex:e")
        return document

File: java-prov/src/main/java/org/example/DefaultNamespace.java
package org.example;

import org.openprovenance.prov.interop.InteropFramework;
import org.openprovenance.prov.model.Entity;
import org.openprovenance.prov.model.Namespace;
import org.openprovenance.prov.vanilla.Document;
import org.openprovenance.prov.vanilla.ProvFactory;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.OutputStream;


public class DefaultNamespace extends TestCase {

    public DefaultNamespace(){
        setFilename("default_namespace");
    }

    @Override
    public Document createDocument() {
        ProvFactory factory = new ProvFactory();
        var document = new org.openprovenance.prov.vanilla.Document();
        var ns = new Namespace();
        ns.addKnownNamespaces();
        ns.registerDefault("https://default.org/");
        ns.register("ex","https://example.org/");
        var e = ns.qualifiedName("ex","e",factory);
        Entity entity = factory.newEntity(e);
        document.getStatementOrBundle().add(entity);
        document.setNamespace(ns);
        return document;
    }
}
